src/unstract/sdk/adapters/x2text/llm_whisperer/src/llm_whisperer.py

```diff
@@ -384,7 +383,7 @@ def _write_output_to_file(
                 metadata_file_name = output_file_path.with_suffix(".json").name
                 metadata_file_path = metadata_dir / metadata_file_name
                 # Ensure the metadata directory exists
-                fs.mkdir(metadata_dir, create_parents=True)
+                fs.mkdir(str(metadata_dir), create_parents=True)
```

Comment:
@harini-venkataraman Wondering if we have to cast metadata_dir to str or follow what we have done here in line 374 ?

Reply:
@gaya3-zipstack `metadata_dir` does not fetch `name` param. Do you mean this?
The way in which I have handled, passes the metadata path appended with the root path - `prompt_studio_data/<org_id>/<user_id>/<tool_id>/` and this works. I actually didnt get the optimiations you refer here.

Reply:
Got it.  This is not Path lib as you mentioned in DM.